### ISSUE TITLE
GoLint problem

### ISSUE THREAD

**Issue by vired** [fatih/vim-go#2830]
### What did you do? (required: The issue will be **closed** when not provided)
i create a file as in https://golang.org/doc/code.html#ImportingLocal and pasted the code
there is a comment before the function declaration. I deleted it. then i run GoLint . it pass and says `no write since last change`. I understand that Golint don't work on a unsaved buffer.

Then i saved the file. Then ran GoLint. This times Fails. I press enter then at the bottom of nvim this comes -
```
  1 reverse.go|5 col 1| exported function ReverseRunes should have comment or be unexported    
[Quickfix List] GoLint                                                      1,1             All
```
then i undo my change to get the comment back then save the file. Then run GoLint. This time passes. but when i press enter the Quickfix List still available. why?

Then try to quit NeoVim. but a unnamed buffer comes with - 
```
reverse.go|5 col 1| exported function ReverseRunes should have comment or be unexported 
```
why This comes up?

### What did you expect to happen?
- it would be nice if i can run GoLint on the current buffer.
- Quickfix List should disappear after GoLint PASS.
- the unnamed buffer should not come when GoLint PASS.
- i think `5:1` is simple than `5 col 1`, if you want to be explicit `line 5 col 1` would be better.
 

### What happened instead?
- i can't run GoLint on a unsaved buffer.
- Quickfix List doesn't disappear after GoLint pass
- before quitting nvim a unnamed buffer comes with quickfix list when golint has passed

### Configuration (**MUST** fill this out):
on my init.vim i have this for vim-go
    let g:go_template_use_pkg = 1

#### vim-go version:
```
running - git log -n 1
commit b995086646f29de25acbe7df2c08cc196c4641d7 (grafted, HEAD -> master, origin/master, origin/HEAD)
```

####  `vimrc` you used to reproduce:

<details><summary>vimrc</summary><br><pre>
call plug#begin('~/.config/nvim/plugged')
Plug 'tpope/vim-surround'
Plug 'preservim/nerdtree'
Plug 'fatih/vim-go'
call plug#end()

" some basics
set nocompatible
set number
syntax on
filetype plugin on
set encoding=utf-8

" key maps

" from https://thoughtbot.com/blog/vim-splits-move-faster-and-more-naturally
nnoremap `<C-J> <C-W><C-J>`
nnoremap `<C-K> <C-W><C-K>`
nnoremap `<C-L> <C-W><C-L>`
nnoremap `<C-H> <C-W><C-H>`

" vim-go
let g:go_template_use_pkg = 1

"" from Luke Smith's void rice

" nerdtree
map <leader>n :NERDTreeToggle<CR>
autocmd bufenter * if (winnr("$") == 1 && exists("b:NERDTree") && b:NERDTree.isTabTree()) | q | endif

" Automatically deletes all trailing whitespace on save.
autocmd BufWritePre * %s/\s\+$//e

" Automatically deletes all newlines end of the line on save; must be run after
" cleaning trailing whitespaces
autocmd BufWritepre * %s/\n\+\%$//e

</pre></details>

#### Vim version (first three lines from `:version`):

NVIM v0.4.3
Build type: RelWithDebInfo
LuaJIT 2.1.0-beta3

#### Go version (`go version`):
go version go1.14.2 linux/amd64

#### Go environment
<details><summary><code>go env</code> Output:</summary><br><pre>
GO111MODULE="on"
GOARCH="amd64"
GOBIN=""
GOCACHE="/home/ahmed/.cache/go-build"
GOENV="/home/ahmed/.config/go/env"
GOEXE=""
GOFLAGS=""
GOHOSTARCH="amd64"
GOHOSTOS="linux"
GOINSECURE=""
GONOPROXY=""
GONOSUMDB=""
GOOS="linux"
GOPATH="/home/ahmed/dl/go"
GOPRIVATE=""
GOPROXY="https://proxy.golang.org,direct"
GOROOT="/usr/local/go"
GOSUMDB="sum.golang.org"
GOTMPDIR=""
GOTOOLDIR="/usr/local/go/pkg/tool/linux_amd64"
GCCGO="gccgo"
AR="ar"
CC="gcc"
CXX="g++"
CGO_ENABLED="1"
GOMOD="/dev/null"
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -fmessage-length=0 -fdebug-prefix-map=/tmp/go-build480167300=/tmp/go-build -gno-record-gcc-switches"

</pre></details>

#### gopls version
<details><summary><code>gopls version</code> Output:</summary><br><pre>
golang.org/x/tools/gopls 0.4.0
    golang.org/x/tools/gopls@v0.4.0 h1:G4+YP9kaV4dJb79J5MobyApxX493Qa6VoiTceUmxqik=

</pre></details>

**Comment by bhcleek** [comment]
> i can't run GoLint on a unsaved buffer.

You only need to `set autowrite` or `set autowriteall` in your vimrc, and vim-go will save the buffer before linting.

> Quickfix List should disappear after GoLint PASS.
> the unnamed buffer should not come when GoLint PASS.

Agreed. PR is up to fix this: #2833. Feel free to try it out now.

> i think 5:1 is simple than 5 col 1, if you want to be explicit line 5 col 1 would be better.

This isn't in vim-go's control; it's how Vim works. You may be able to control it through some options or an autocmd that works with the qf buffer type, though.

**Comment by bhcleek** [comment]
Fixed by #2833
